domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1802,59 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// AllModelActivationStatusQuery returns a SQL statement that will return all model activated status.
+func (st *State) AllModelActivationStatusQuery() string {
+	return "SELECT activated from model"
+}
+
+// GetActivatedModelUUIDs returns the UUIDs of all activated models from the given list of UUIDs.
+//
+// Returns:
+//   - []string: A subset of the input UUIDs containing only those of activated models.
+//   - error: An error if the retrieval process fails.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []string) ([]string, error) {
```

Comment:
We should prefer strong types where possible. Can the return of this func be a slice of `coremodel.UUID` ?